Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs

```diff
@@ -517,11 +517,42 @@ public override void Update()
             if (!IsSetupValid) { return; }
 
             UpdatePointers();
+
+            if (gazeProviderPointingData?.Pointer != null)
+            {
+                UpdateGazeProvider();
+            }
             UpdateFocusedObjects();
 
             PrimaryPointer = primaryPointerSelector?.Update();
         }
 
+        /// <summary>
+        /// Update the hit information for the gaze provider.
+        /// 
+        /// For the sake of StabilizationPlaneModifier and potentially other 
+        /// components that care where the user's looking, we need to do a gaze 
+        /// raycast even if gaze isn't used for focus.
+        /// Assumes gaze provider is not null;
+        /// </summary>
+        private void UpdateGazeProvider()
+        {
+            // We only update if interaction is not enabled because if interaction is enabled 
+            // for this pointer, we have already updated hit data
```

Comment:
> we have already updated hit data [](start = 32, length = 33)

Is there any way to assert that we aren't recalc-ing things? This introduces cross-code dependency that is implicit, meaning we could "fix" this issue in another location and not know to update this (i.e. delete it) later.

Reply:
I looked into whether we could null out and then read some fields, but didn't find any. Instead I refactored the code so that we only ever populate the gazeprovider data in one location -- from the focus provider. Previously we populated the gaze provider data in two locations--once from FocusProvider, once from GazeProvider.InternalGazePointer. Is there still a cross-code dependency?